### PR TITLE
Cluster api feature (fate#323437)

### PR DESCRIPTION
**Pull request by ayoub-belarbi**
This feature is not finished yet and not ready for review.

### CODE REVIEW

**Comment by krauselukas** [comment]
@ayoub-belarbi i was able to check it out today, really nice. i think i'm able to support you with this :)

**Comment by krauselukas** [comment]
@ayoub-belarbi I opened a pull request to your fork of hawk related to this PR

**Comment by ayoub-belarbi** [comment]
> @ayoub-belarbi i was able to check it out today, really nice. i think i'm able to support you with this :)

T

> @ayoub-belarbi i was able to check it out today, really nice. i think i'm able to support you with this :)

Thanks Lucas ;) It will be wonderful if we maintain the api together :)

**Comment by ayoub-belarbi** [comment]
> @ayoub-belarbi I opened a pull request to your fork of hawk related to this PR

It looks good actually :+1:

File: hawk/app/controllers/api/v1/api_controller.rb [hunk 1]
```diff
@@ -25,31 +25,46 @@ def expired(expiry_date)
...
+                    render_expired # Prevent access when the token is expired
+                  else
+                    return true # Authenticated successfully
+                  end
+                else
+                  render_invalid_token # Token invalid, or store is invalid
                 end
               end
+            else
+              render_registration_required # The client needs to register (no store is found)
```
Comment (krauselukas) [inline hawk/app/controllers/api/v1/api_controller.rb]: @ayoub-belarbi LGTM, there is just this edge case where a registration is required even when the file exists. But way cleaner now :fireworks:

**Pull request merged**